### PR TITLE
Issue/425 course todo items

### LINKED ISSUE #425: Implement course TODO items
**What resource needs additional coverage?**
Courses

**What endpoints need to be covered?**
Get course TODO items
[Canvas API: Courses](https://canvas.instructure.com/doc/api/courses.html#method.courses.todo_items)

### PR DESCRIPTION
**Pull request by onomou**
# Proposed Changes

*Implement `course.get_todo_items()`
*Canvas API documentation: https://canvas.instructure.com/doc/api/courses.html#method.courses.todo_items

Fixes #425 .

### CODE REVIEW

File: canvasapi/course.py [hunk 1]
```diff
@@ -2087,6 +2087,21 @@ def get_tabs(self, **kwargs):
             _kwargs=combine_kwargs(**kwargs),
         )
 
+    def get_todo_items(self):
```
[the remark this answers is not in the document]
Comment (onomou) [reply canvasapi/course.py:2090]: I noticed most of the other calls have that, but the course TODO endpoint does not allow any arguments. I compared it to the user TODO endpoint, which does allow one, and thought I should not include it.

[the remark this answers is not in the document]
Comment (onomou) [reply canvasapi/course.py:2090]: Now I see Travis CI complaining about that. It passed the tests on my computer, so I thought everything was fine. I have much to learn...

[the remark this answers is not in the document]
Comment (jonespm) [reply canvasapi/course.py:2090]: Yeah, I guess they added this "fix_missing_kwargs" so it will block any changes without it!

**Comment by jessemcbride** [comment]
We recently got nipped by [a bug with a Course Analytics endpoint that returned JSON](https://github.com/ucfopen/canvasapi/issues/429) instead of a PaginatedList. While I hope Canvas TODOs won't exceed a hundred elements, I don't know if we can guarantee that after looking at the API docs (https://canvas.instructure.com/doc/api/users.html#method.users.todo_items returns a paginated list).

Can we update this to include a new `CanvasObject` type for `TodoItem`, returning results wrapped in a `PaginatedList` to make sure pagination is handled correctly in case there ever are more than a hundred elements?

**Comment by onomou** [comment]
I hope I wrapped `course.get_todo_items()` in a `PaginatedList` correctly. It seems to work on my computer, at least. This will also need to be done with `canvas.get_todo_items()`.

File: canvasapi/todo.py [hunk 2]
```diff
@@ -0,0 +1,6 @@
+from canvasapi.canvas_object import CanvasObject
+
+
+class Todo(CanvasObject):
+    def __str__(self):
+        return "Todo List ({})".format(self.id)
```
Comment (Thetwam) [inline canvasapi/todo.py:6]: `id` isn't a valid attribute for a todo item object. It looks like `type` is the closest thing to a meaningful attribute that is readily available, unless we try for digging deeper for `quiz` or `assignment`.

This `__str__` method is also lacking testing.

I'll do a quick patch

**Comment by Thetwam** [comment]
Looks good, @onomou! Thanks for the contribution! 😁

**Review by Thetwam — DISMISSED** [review]

**Review by Thetwam — APPROVED** [review]
Travis seems to be having some trouble, but all the tests are passing on my end, and not much has changed since it last passed. Just gonna merge this bad boi in.

**Pull request merged**